packages/build/src/homebrew/generate-formula.ts

```diff
@@ -1,28 +1,35 @@
+import * as semver from 'semver';
+import { GithubRepo } from '../github-repo';
 
-export function generateFormula(context: { version: string, sha: string }): string {
-  return `require "language/node"
+export async function generateUpdatedFormula(
+  context: { version: string, sha: string },
+  homebrewCoreFork: GithubRepo
+): Promise<string | null> {
+  const currentFormula = await homebrewCoreFork.getFileContent('Formula/mongosh.rb', 'master');
 
-class Mongosh < Formula
-  desc "The MongoDB Shell"
+  const urlMatch = /url \"([^"]+)\"/g.exec(currentFormula.content);
+  const shaMatch = /sha256 \"([^"]+)\"/g.exec(currentFormula.content);
 
-  homepage "https://github.com/mongodb-js/mongosh#readme"
-  url "https://registry.npmjs.org/@mongosh/cli-repl/-/cli-repl-${context.version}.tgz"
-  version "${context.version}"
+  if (!urlMatch || !shaMatch) {
+    throw new Error('mongosh: could not find url or sha field in homebrew/core formula');
+  }
 
-  # This is the checksum of the archive. Can be obtained with:
-  # curl -s https://registry.npmjs.org/@mongosh/cli-repl/-/cli-repl-${context.version}.tgz | shasum -a 256
-  sha256 "${context.sha}"
+  const currentUrl = urlMatch[1];
+  const currentSha = shaMatch[1];
 
-  depends_on "node@14"
+  const newUrl = `https://registry.npmjs.org/@mongosh/cli-repl/-/cli-repl-${context.version}.tgz`;
```

Comment:
Do we maybe want to use the non-scoped `mongosh` npm package? (Also obviously unrelated to the existing code here)

Reply:
We can definitely do that... I'd probably suggest for this move that we provide a separate manual PR first to switch the formula structure in general and get it approved. Version bumps should really just be URL with version change and the hash change I assume...

Reply:
Hello, Homebrew maintainer here. The automated pull request to update the formula (Homebrew/homebrew-core#79146) doesn't have a PR description/body, so that makes the `request-info` bot unhappy. I also see from this PR that you have a formula template instead of updating just the URL and checksum.

You could use a GitHub Action instead to update the formula, like [`action-homebrew-bump-formula`](https://github.com/dawidd6/action-homebrew-bump-formula) (the author, `dawidd6`, is a Homebrew maintainer). If you wouldn't want to use a 3rd party GitHub Action, just a simple workflow which does `brew bump-formula-pr` would be good too. (I'd be willing to submit a PR for this.)

Reply:
Thanks for the comment @nandahkrishna :) The formula template is only for the test - the code here essentially grabs the latest formula from `homebrew/homebrew-core` and then replaces the URL as well as the SHA by using two regular expressions.

Would that workflow be fine? Right now this is tied to a workflow that runs in the Evergreen CI.

An alternative we could do is that we set up a GitHub Action that is triggered by publishing a new release on GitHub (a non-draft release). This could then use the suggested action.

What do you think @addaleax? 😊

Reply:
> An alternative we could do is that we set up a GitHub Action that is triggered by publishing a new release on GitHub (a non-draft release). This could then use the suggested action.

I have to admit that it doesn’t feel *super* clean to me to mix two different CI systems in a way where they depend on each other…

I’m not sure, but it seems like the main concern here really is the missing PR description, I think just addressing that is fine – it should look the same from the homebrew maintainer’s perspective, right?

Reply:
> Would that workflow be fine?

Thanks for the clarification, that would be fine. I incorrectly assumed that the template was being used to update the formula, in which case there would've been the risk of using an outdated template (in case there were updates in `homebrew/core` that weren't also updated here).

> it seems like the main concern here really is the missing PR description, I think just addressing that is fine – it should look the same from the homebrew maintainer’s perspective, right?

Yes, this should be good in my opinion.